token-metadata/program/src/processor/metadata/transfer.rs

```diff
@@ -313,6 +313,25 @@ fn transfer_v1(program_id: &Pubkey, ctx: Context<Transfer>, args: TransferArgs)
             // validates the derivation
             assert_keys_equal(&new_pda_key, destination_token_record_info.key)?;
 
+            // We need to check whether the destination token account has a delegate set. If it does,
+            // we do not allow the transfer to proceed since we do not know the type of the delegate
+            // to complete the information on the token record.
+            let destination_token =
+                Account::unpack(&ctx.accounts.destination_info.try_borrow_data()?)?;
+
+            if let COption::Some(delegate) = destination_token.delegate {
+                if destination_token_record_info.data_is_empty() {
+                    return Err(MetadataError::DelegateAlreadyExists.into());
```

Comment:
It could be, but trying to avoid adding another error to save space. So if you get a `DelegateAlreadyExists` error on transfer, this can only be triggered by a delegate already set on the destination token account. But agree that have a specific error for this would be better.

cc: @samuelvanderwaal

Reply:
Sounds good to me